subcommittee/coding-guidelines/initiatives/safety-critical-rust-coding-guidelines/src/coding-guidelines/types-and-traits.rst

```diff
@@ -0,0 +1,54 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Coding Guidelines Subcommittee Contributors
+
+.. default-domain:: coding-guidelines
+
+Types and Traits
+================
+
+.. guideline:: Avoid Implicit Integer Wrapping
+   :id: gui_xztNdXA2oFNB
+   :status: draft
+   :fls: fls_cokwseo3nnr
+   :tags: numerics
+   :category: types
+   :recommendation: required
+
+   Code must not rely on Rust's implicit integer wrapping behavior that occurs in release builds. 
+   Instead, explicitly handle potential overflows using the standard library's checked, 
+   saturating, or wrapping operations.
```

Comment:
or ensure integer operations don't overflow.  I don't know rust very well, but if you have (int i = 0, i < 10; i++) there is not need to explicitly handle overflows because none are possible.